### PR TITLE
Allow choosing a frame at take pic, make frame and font an option

### PR DESCRIPTION
**Pull request by andi34**

### CODE REVIEW

File: api/print.php [hunk 1]
```diff
@@ -51,9 +55,9 @@
         $source = imagecreatefromjpeg($filename_source);
         $code = imagecreatefrompng($filename_codes);
 
-        if ($config['print_frame'] == true) {
+        if ($config['print_frame'] == true && !($config['take_frame'])) {
```
Comment (sualko) [inline api/print.php]: Not related to this pr, but we can remove ` == true`.

File: api/print.php [hunk 2]
```diff
@@ -51,9 +55,9 @@
...
             $print = imagecreatefromjpeg($filename_source);
-            $rahmen = @imagecreatefrompng('../resources/img/frames/frame.png');
+            $rahmen = @imagecreatefrompng($print_frame);
```
Comment (sualko) [inline api/print.php]: Is there a reason you suppress errors? It makes it harder to debug an issue.

File: lib/collage.php [hunk 3]
```diff
@@ -22,6 +22,14 @@ function createCollage($srcImagePaths, $destImagePath) {
 
         $tempSubImage = imagecreatefromjpeg($srcImagePaths[$i]);
 
+        if ($takeFrame) {
```
Comment (sualko) [inline lib/collage.php]: Is this not the same code as in `api/applyEffects.php`?

File: lib/resize.php [hunk 4]
```diff
@@ -15,4 +15,23 @@ function resizeImage($image, $max_width, $max_height)
     $new_height = ceil($scale*$old_height);
 
     return imagescale($image, $new_width, $new_height, IMG_TRIANGLE);
-}
\ No newline at end of file
+}
+
+function ResizePngImage($image, $max_width, $max_height)
```
Comment (sualko) [inline lib/resize.php]: Uppercase names indicates usually classes, so this is irritating.

Comment (andi34) [reply lib/resize.php]: Copy paste from print.php, was about to call it from here too there. But sure we can update that.

File: api/print.php (hunk 1 above)
Comment (andi34) [reply api/print.php]: We can do that here

File: api/print.php (hunk 2 above)
Comment (andi34) [reply api/print.php]: Copy paste from print.php

Comment (andi34) [reply api/print.php]: It was there already

File: lib/collage.php (hunk 3 above)
Comment (andi34) [reply lib/collage.php]: Didn't got it working from a central point...

File: api/print.php (hunk 1 above)
Comment (andi34) [reply api/print.php]: done

File: lib/collage.php (hunk 3 above)
Comment (andi34) [reply lib/collage.php]: Seems like this is left. There's some duplicate code to apply a frame which maybe you or someone else is able to centralize? I tried it but failed on that... :disappointed:

File: api/applyEffects.php [hunk 5]
```diff
@@ -73,6 +73,14 @@
     $imageResource = effectPolaroid($imageResource, $polaroid_rotation, 200, 200, 200);
 }
 
+if ($config['take_frame'] && $_POST['isCollage'] !== 'true') {
+    $rahmen = imagecreatefrompng($config['take_frame_path']);
```
Comment (sualko) [inline api/applyEffects.php]: I think `rahmen` is a german word and should be renamed to `frame`.

**Comment by andi34** [comment]
Guess it is good to go at the moment, maybe in future someone can move this to be called from a central place to avoid having the same code many times.

File: lib/collage.php (hunk 3 above)
Comment (andi34) [reply lib/collage.php]: can be optimized in future

**Pull request merged**